### PR TITLE
added library code

### PR DESCRIPTION
**Pull request by Simon-Heck**

### CODE REVIEW

File: README copy.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Scripts for running unrealCV and fastAI with UE5
```
Comment (anthonyjclark) [inline README copy.md]: What is this file?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from setuptools import setup
+setup(name="ue5env",version="0.1",description="wrapper for unreal engine using unrealcv",author="Simon Heck",license="MIT",packages=["ue5env"],zip_safe=False)
```
Comment (anthonyjclark) [inline setup.py]: This file should be hit with a code formatter.

File: ue5env/__init__.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
```
Comment (anthonyjclark) [inline ue5env/__init__.py]: Remove this. This file is not intended to be executed as a script.

File: ue5env/__init__.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+import shutil
```
Comment (anthonyjclark) [inline ue5env/__init__.py:1]: Prefer `pathlib` to `shutil`

File: ue5env/__init__.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+__author__ = "Simon Heck"
+
+
+class UE5EnvWrapper:
+    """Wrapper that handles interactions between the program and Unreal Engine and takes the forwarded port as an input"""
+
+    def __init__(self, port: int = 8500):
+        global ue5
+        ue5 = unrealcv.Client(("localhost", port))
+        ue5.connect(5)
```
Comment (anthonyjclark) [inline ue5env/__init__.py]:
```suggestion
        ue5.connect(timeout=5)
```

File: ue5env/__init__.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        if ue5.isconnected():
+            print(ue5.request("vget /unrealcv/status"))
+        else:
+            print("Failed to Connect to UnrealCV server")
+            exit(0)
```
Comment (anthonyjclark) [inline ue5env/__init__.py]: Raise an exception instead of exiting.

File: ue5env/__init__.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+    def isconnected(self):
+        """Is the program connected to Unreal"""
+        return self.connected
```
Comment (anthonyjclark) [inline ue5env/__init__.py]: Should this be:
```suggestion
        return self.ue5.isconnected()
```

File: ue5env/__init__.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+    def reset(self):
+        """Reset robot to start location. Interacts with UE5 Blueprint."""
+        ue5.request(f"vset /action/keyboard backspace 1")
+
+    def getCameraLocation(self, cameraID: float) -> tuple[float, float, float]:
```
Comment (anthonyjclark) [inline ue5env/__init__.py]: Should camera id be a float or an int? Also, should we default it to `0`?

File: ue5env/__init__.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        """Returns X, Y, Z location of a camera in the Unreal Environment."""
+        x, y, z = ue5.request(f"vget /camera/{cameraID}/location").split()
+        return (float(x), float(y), float(z))
```
Comment (anthonyjclark) [inline ue5env/__init__.py]:
```suggestion
        return float(x), float(y), float(z)
```

File: ue5env/__init__.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+    def getCameraRotation(self, cameraID: float) -> tuple[float, float, float]:
+        """Returns Pitch, Yaw, and Roll values for Camera number"""
+        pitch, yaw, roll = ue5.request(f"vget /camera/{cameraID}/rotation").split()
+        return float(pitch), float(yaw), float(roll)
+
+    def left(self, degreeRot: float, cameraNum: int) -> None:
```
Comment (anthonyjclark) [inline ue5env/__init__.py]: How is `cameraNum` different from `cameraID`?

**Review by anthonyjclark — CHANGES_REQUESTED** [review]

File: ue5env/__init__.py (hunk 4 above)
Comment (Simon-Heck) [reply ue5env/__init__.py:1]: Marked with a @TODO. Not sure where unrealCV saves images when working with the remote server and this method is not needed until datacollection starts

**Pull request closed without merging**